server/src/main/java/org/opensearch/extensions/RegisterTransportActionsRequest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Request to register extension Transport actions
+ *
+ * @opensearch.internal
+ */
+public class RegisterTransportActionsRequest extends TransportRequest {
```

Comment:
Need a class to test this.  (Should be able to use object equality after writing and reading to a stream.)